crates/cdk/src/nuts/nut06.rs

```diff
@@ -79,6 +81,20 @@ pub struct MintInfo {
     /// message of the day that the wallet must display to the user
     #[serde(skip_serializing_if = "Option::is_none")]
     pub motd: Option<String>,
+    /// server unix timestamp
+    #[serde(serialize_with = "serialize_time")]
+    pub time: Option<u64>,
+}
+
+fn serialize_time<S>(_: &Option<u64>, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let epoch = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .unwrap()
+        .as_secs();
+    serializer.serialize_u64(epoch)
```

Comment:
I don't think we should use a serialization function to change or add data only to serialize existing data. So I actually prefer your first approach.